### ISSUE TITLE
MongoDB chart does not run in OpenShift

### ISSUE THREAD

**Issue by migmartri** [bitnami/charts#932]
MongoDB chart installation seems to fail due to the securityContext in OpenShift

```bash
$ oc version
oc v3.9.41
kubernetes v1.9.1+a0ce1bc657
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server REDACTED
openshift v3.9.33
kubernetes v1.9.1+a0ce1bc657
```

```
$  helm install -n miguel-mongodb bitnami/mongodb

$  helm list
NAME          	REVISION	UPDATED                 	STATUS  	CHART        	APP VERSION	NAMESPACE
miguel-mongodb	1       	Mon Nov 19 11:36:25 2018	DEPLOYED	mongodb-4.9.0	4.0.3      	miguel   


$ kubectl get pods
=> none

$ kubectl describe rs miguel-mongodb-6d94445864

...
Type     Reason        Age                       From                   Message
  ----     ------        ----                      ----                   -------
  Warning  FailedCreate  <invalid> (x16 over 99s)  replicaset-controller  Error creating: pods "miguel-mongodb-6d94445864-" is forbidden: unable to validate against any security context constraint: [fsGroup: Invalid value: []int64{1001}: 1001 is not an allowed group spec.containers[0].securityContext.securityContext.runAsUser: Invalid value: 1001: must be in the ranges: [1000140000, 1000149999]]
```

The error returned is:

>  Invalid value: 1001: must be in the ranges: [1000140000, 1000149999]]

and it seems to be related to the fact that we set securityContext in our values.yaml file. If I remove or override that section, it seems to work fine.

https://github.com/bitnami/charts/blob/master/upstreamed/mongodb/values.yaml#L65-L68

**Comment by migmartri** [comment]
Maybe related https://github.com/bitnami/charts/issues/827

**Comment by tompizmor** [comment]
You are right. By default, the MongoDB chart fails to install on Openshift. 

In order to make it work, you need to specify a couple of parameters regarding `securityContext` 

This command should work:

```
helm install bitnami/mongodb --set securityContext.fsGroup="",securityContext.runAsUser=1000140001
```

**Comment by migmartri** [comment]
Thanks @tompizmor for the answer. In our tests, we also found the workaround of not setting the security context at all https://github.com/kubeapps/kubeapps/issues/676#issuecomment-436015118 and still MongoDB seems to run as non-root (as expected)

My question would be, does it make sense to enforce the securityContext moreover taking into account that it makes it fail in OpenShift? 

Thanks!

**Comment by tompizmor** [comment]
I think we could get rid of `securityContext.runAsUser`, however, if the container is non-root by default, we need something to change the permissions in the persistent volume so we still need `securityContext.fsGroup` for standard kubernetes distributions.

**Comment by migmartri** [comment]
@tompizmor that makes sense, thanks for the detailed info.

I don't have experience on this but just in case I throw the idea. Do you think that advance features as supplemental groups ranges could help here to make it work in both regular k8s and openshift? https://docs.openshift.com/enterprise/3.1/install_config/persistent_storage/pod_security_context.html#fsgroup.

Also, do you know why the uuid is different in OS? 1000140001 is this a convention?

Thanks!

**Comment by tompizmor** [comment]
I didn't know about `supplementalGroups' either but maybe is worth it a bit of investigation.

> Also, do you know why the uuid is different in OS? 1000140001 is this a convention?

AFAIK it is different for each project that you create
